sdk/cosmosdb/cosmos/src/queryExecutionContext/parallelQueryExecutionContextBase.ts

```diff
@@ -376,11 +388,11 @@ export abstract class ParallelQueryExecutionContextBase implements ExecutionCont
               });
             }
           } catch (err: any) {
-            this.err = new Error(
+            this.err = new CosmosException(
               `Extracted DocumentProducer from the priority queue fails to get the \
-                                    buffered item. Due to ${JSON.stringify(err)}`
+                                    buffered item. Due to ${err}`
             );
-            this.err.headers = this._getAndResetActiveResponseHeaders();
+            new CosmosException(this._getAndResetActiveResponseHeaders());
```

Comment:
Missing assignment of this `new CosmosException(this._getAndResetActiveResponseHeaders());`